web/src/components/gitops/SetupProvider.jsx

```diff
@@ -0,0 +1,269 @@
+                <img src={warning} alt="warning" />
+                <p>Repository access needed</p>
+              </div>
+            ) : (
+              gitopsEnabled &&
+              gitopsConnected && (
+                <div className="flex" style={{ gap: "5px", color: "green" }}>
+                  <img src={enabled} alt="enabled" />
+                  <p>Enabled</p>
+                </div>
+              )
+            )}
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  const downstream = app?.downstream;
+  const gitops = downstream?.gitops;
+  const gitopsEnabled = gitops?.enabled;
+  const gitopsConnected = gitops?.isConnected;
+
+  return (
+    <div
+      key={`${step.step}-active`}
+      className="GitOpsDeploy--step u-textAlign--left"
+    >
+      <p className="step-title">{step.title}</p>
+      <p className="step-sub">
+        Connect a git version control system so all application updates are
+        committed to a git repository. When GitOps is enabled, you cannot deploy
+        updates directly from the admin console.
+      </p>
+      <div className="flex-column u-textAlign--left ">
+        <div className="flex alignItems--center u-marginBottom--30">
+          {isSingleApp && app ? (
+            <div className="u-marginRight--5">{getLabel(app)}</div>
+          ) : (
+            <div className="flex flex1 flex-column u-marginRight--10">
+              <p className="u-fontSize--large u-textColor--primary u-fontWeight--bold u-lineHeight--normal">
+                Select an application to configure
+              </p>
+              <div className="u-position--relative u-marginTop--5 u-marginBottom--10">
+                <Select
+                  className="replicated-select-container select-large "
+                  classNamePrefix="replicated-select"
+                  placeholder="Select an application"
+                  options={apps}
+                  isSearchable={false}
+                  getOptionLabel={(app) => getLabel(app)}
+                  // getOptionValue={(app) => app.label}
+                  value={selectedApp}
+                  onChange={handleAppChange}
+                  isOptionSelected={(option) => {
+                    option.value === selectedApp;
+                  }}
+                />
+              </div>
+            </div>
+          )}
+          <div className="flex flex1 flex-column u-fontSize--small u-marginTop--20">
+            {gitopsEnabled && gitopsConnected && (
+              <a
+                style={{ color: "blue", cursor: "pointer" }}
+                disabled={disablingGitOps}
+                onClick={promptToDisableGitOps}
+              >
+                {disablingGitOps
+                  ? "Disabling GitOps"
+                  : "Disable GitOps for this app"}
+              </a>
+            )}
+          </div>
+        </div>
+        {renderGitOpsProviderSelector({
+          owner,
```

Comment:
I initially thought I could take everything out and put it into this component but it turns out that alot of data is being shared. so this renderGitOpsProviderSelector is actually rendered in the parent component. I will be using Context API to make the code better in the next PR.